src/Symfony/Component/Validator/ConstraintValidatorFactory.php

```diff
@@ -42,7 +39,7 @@ public function getInstance(Constraint $constraint)
 
         if (!isset($this->validators[$className])) {
             $this->validators[$className] = 'validator.expression' === $className
```

Comment:
If it should be changed, what about the service definition in validator.xml, see above?